### PR TITLE
(FM-8856) Convert acceptance testing to litmus

### PR DESCRIPTION
**Pull request by michaeltlombardi**
This commit changes the acceptance testing framework from beaker to
litmus, enabling acceptance testing in CI.

### CODE REVIEW

**Comment by michaeltlombardi** [comment]
TODO: update travis/appveyor to run litmus tests, which will not pass until Litmus 0.14.0 is released (they depend on updated functionality in `spec_helper_acceptance`).

Tested locally with litmus from master, windows passed, still waiting on *nix - will investigate in the morning as it's approaching 0200.

File: spec/spec_helper_acceptance_local.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                  File.join(script_folder, "rhel_#{os[:release].to_i}.sh")
+                when 'debian'
+                  File.join(script_folder, "debian_#{os[:release].to_i}.sh")
+                when 'darwin'
+                  File.join(script_folder, "darwin.sh")
+                when 'windows'
+                  File.join(script_folder, "windows.ps1")
+                end
+  Helper.instance.bolt_run_script(script_path)
+end
```
Comment (glennsarti) [inline spec/spec_helper_acceptance_local.rb]: Should verify here `raise 'where u at pwsh' unless pwsh_installed?`

File: spec/spec_helper_acceptance_local.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+              'apt-get remove powershell -y'
+            when 'redhat'
+              'yum remove powershell -y'
+            when 'darwin'
+              'brew cask uninstall powershell'
+            when 'windows'
+              interpolate_powershell('Get-Command pwsh | Select-Object -ExpandProperty Source | ForEach-Object { Remove-Item -Path (Split-Path -Parent $_) -Recurse -Force }')
+            end
+  Helper.instance.run_shell(command)
+end
```
Comment (glennsarti) [inline spec/spec_helper_acceptance_local.rb]: Should verify here `raise 'just go away!!!' if pwsh_installed?`

**Comment by michaeltlombardi** [comment]
GitHub runner added.

**Comment by david22swan** [comment]
If it's not to much trouble could you implement the changes shown in this pr please: https://github.com/puppetlabs/puppetlabs-accounts/pull/293

**Comment by david22swan** [comment]
Also could you push your changes to the release branch temporarily to check that the github actions work as they should

**Comment by michaeltlombardi** [comment]
Oh sweet I didn't know we had templated out the provision file! Will update and push to release :)

File: spec/acceptance/exec_pwsh_spec.rb [hunk 3]
```diff
@@ -1,56 +1,77 @@
...
+    }
 
-describe 'pwsh provider:', :if => powershell6_agents.count > 0 do
-  def platform_string(host, windows, posix)
-    if windows_platform?(host)
-      windows
-    else
-      posix
+    it 'errors because pwsh is not in the path' do
+      fail 'noodle' unless pwsh_installed? == false
```
[the remark this answers is not in the document]
Comment (michaeltlombardi) [reply spec/acceptance/exec_pwsh_spec.rb]: 😂 I'm surprised and impressed with myself at how work safe that placeholder error is.

File: spec/acceptance/exec_pwsh_spec.rb [hunk 4]
```diff
@@ -1,56 +1,77 @@
...
+  context 'when pwsh is not installed' do
+    before(:all) do
+      uninstall_pwsh if pwsh_installed?
+      raise 'failed to remove pwsh' if pwsh_installed?
+    end
+
+    let(:manifest) {
+      <<-MANIFEST
+        exec{'TestPowershell':
+          command   => 'Get-Process > /process.txt',
```
[the remark this answers is not in the document]
Comment (michaeltlombardi) [reply spec/acceptance/exec_pwsh_spec.rb]: Ahh, this one is going to blow up regardless as it's in the negative test case - and if it _doesn't_ blow up for the **right** reason we'll catch that.

File: spec/acceptance/exec_pwsh_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (cmccrisken-puppet) [reply spec/acceptance/exec_pwsh_spec.rb]: To be fair, if I saw "Noodle!", that would peak my interest.

**Comment by michaeltlombardi** [comment]
Ignore "failing" PR test for appveyor... it's pointing to #288 for some unknown reason, not this PR.

File: spec/spec_helper_acceptance_local.rb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+def pwsh_installed?
+  Helper.instance.run_shell('pwsh -v', expect_failures: true).exit_code == 0
+end
+
+def cleanup_files
+  absent_files_manifest = if os[:family] == 'windows'
+                            'file{["c:/services.txt","c:/process.txt","c:/try_success.txt","c:/catch_shouldntexist.txt","c:/try_shouldntexist.txt","c:/catch_success.txt"]: ensure => absent }'
+                          else
+                            'file{["/services.txt","/process.txt","/try_success.txt","/catch_shouldntexist.txt","/try_shouldntexist.txt","/catch_success.txt"]: ensure => absent }'
```
Comment (cmccrisken-puppet) [inline spec/spec_helper_acceptance_local.rb]: Thought I'd commented on this in my previous review, but the comment isn't there...odd. 

All these files bar `/process.txt` are outputting to `/tmp` now - we may want to clean them up explicitly. I know `/tmp` will get cleared eventually, but I reckon these files will stick around if the VM remained powered on and a second test run was kicked off.

File: spec/spec_helper_acceptance_local.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+def localhost_windows?
+  os[:family] == 'windows' && ENV['TARGET_HOST'] == 'localhost'
+end
+
+RSpec.configure do |c|
+  c.before :suite do
+    Helper.instance.run_shell('puppet module install puppetlabs/pwshlib')
+    cleanup_files
+  end
+end
```
Comment (cmccrisken-puppet) [inline spec/spec_helper_acceptance_local.rb]: nit: Add newline to end of file.

File: spec/spec_helper_acceptance_local.rb (hunk 5 above)
Comment (michaeltlombardi) [reply spec/spec_helper_acceptance_local.rb]: This is a good catch, these paths needed to be updated - this method should do exactly what you're describing: ensure the files are removed prior to the test suite execution.

**Pull request merged**